Description: Update symbols.
Origin: upstream, commit:1143
---
 src/libmtp.sym |   13 +++++++++++++
 1 file changed, 13 insertions(+)

--- libmtp.orig/src/libmtp.sym
+++ libmtp/src/libmtp.sym
@@ -27,6 +27,19 @@ LIBMTP_Clear_Errorstack
 LIBMTP_Dump_Errorstack
 LIBMTP_Get_Storage
 LIBMTP_Format_Storage
+LIBMTP_Get_String_From_Object
+LIBMTP_Get_u64_From_Object
+LIBMTP_Get_u32_From_Object
+LIBMTP_Get_u16_From_Object
+LIBMTP_Get_u8_From_Object
+LIBMTP_Set_Object_String
+LIBMTP_Set_Object_u32
+LIBMTP_Set_Object_u16
+LIBMTP_Set_Object_u8
+LIBMTP_Get_Property_Description
+LIBMTP_Is_Property_Supported
+LIBMTP_Get_Allowed_Property_Values
+LIBMTP_destroy_allowed_values_t
 LIBMTP_new_file_t
 LIBMTP_destroy_file_t
 LIBMTP_Get_Filetype_Description
